plugin/storage/grpc/shared/grpc_server.go

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) 2019 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shared
+
+import (
+	"context"
+
+	"github.com/jaegertracing/jaeger/model"
+	"github.com/jaegertracing/jaeger/proto-gen/storage_v1"
+	"github.com/jaegertracing/jaeger/storage/spanstore"
+)
+
+const spanBatchSize = 1000
+
+// GRPCServer implements shared.StoragePlugin and reads/writes spans and dependencies
+type GRPCServer struct {
+	Impl StoragePlugin
+}
+
+// GetDependencies returns all interservice dependencies
+func (s *GRPCServer) GetDependencies(ctx context.Context, r *storage_v1.GetDependenciesRequest) (*storage_v1.GetDependenciesResponse, error) {
+	deps, err := s.Impl.GetDependencies(r.EndTime, r.EndTime.Sub(r.StartTime))
+	if err != nil {
+		return nil, err
+	}
+	return &storage_v1.GetDependenciesResponse{
+		Dependencies: deps,
+	}, nil
+}
+
+// WriteSpan saves the span
+func (s *GRPCServer) WriteSpan(ctx context.Context, r *storage_v1.WriteSpanRequest) (*storage_v1.WriteSpanResponse, error) {
+	err := s.Impl.WriteSpan(r.Span)
+	if err != nil {
+		return nil, err
+	}
+	return &storage_v1.WriteSpanResponse{}, nil
+}
+
+// GetTrace takes a traceID and streams a Trace associated with that traceID
+func (s *GRPCServer) GetTrace(r *storage_v1.GetTraceRequest, stream storage_v1.SpanReaderPlugin_GetTraceServer) error {
+	trace, err := s.Impl.GetTrace(stream.Context(), r.TraceID)
+	if err != nil {
+		return err
+	}
+
+	var allSpans [][]model.Span
```

Comment:
It's not necessary to accumulate before sending output. Perhaps can use approach similar to query-service which reuses the helper method for GetTrace/FindTraces

https://github.com/jaegertracing/jaeger/blob/90a1668f7693f739504c63d54affc8089cd6aa83/cmd/query/app/grpc_handler.go#L60